### PR TITLE
Stop recommending `typed: strong`

### PR DESCRIPTION
**Pull request by elliottt**
### Motivation

`typed: strong` is very fragile and generally unsupported. As such, we should stop recommending it when using `--suggest-typed`.

### Test plan

See included automated tests.

### CODE REVIEW

File: main/realmain.cc [hunk 1]
```diff
@@ -104,6 +104,22 @@ core::StrictLevel levelMinusOne(core::StrictLevel level) {
...
+        case core::StrictLevel::Ignore:
+        case core::StrictLevel::False:
+        case core::StrictLevel::True:
+        case core::StrictLevel::Strict:
+            return level;
+        case core::StrictLevel::Strong:
+        case core::StrictLevel::Max:
+            return core::StrictLevel::Strict;
+        default:
+            Exception::raise("Should never happen");
```
Comment (jez) [inline main/realmain.cc:119]: If you delete the `default` case, clang will check the exhaustiveness for you.

File: test/cli/suggest-typed-true/suggest-typed-true.out [hunk 2]
```diff
@@ -157,8 +157,16 @@ class Foo
...
 No errors! Great job.
 -------------------------
-No errors! Great job.
-# typed: strong
+suggest-typed-and-type.rb:1: You could add `# typed: strict` https://srb.help/7022
+     1 |# typed: strong
+     2 |module I1
+  Autocorrect: Done
+    suggest-typed-and-type.rb:1: Replaced with `# typed: strict
+`
```
Comment (jez) [inline test/cli/suggest-typed-true/suggest-typed-true.out:165]: Do you have any idea where this newline is coming from?

Comment (jez) [reply test/cli/suggest-typed-true/suggest-typed-true.out:165]: I guess maybe we unconditionally suggest putting a newline, and there just didn't used to be a suggestion here. 👍 nvm, convinced myself

File: main/realmain.cc (hunk 1 above)
Comment (elliottt) [reply main/realmain.cc:119]: I gave this a try, and the remaining StrictLevel values were `Internal`, `None`, `Autogenerated`, and `StdLib`. Should we just map them back to themselves for the purposes of recommendation?

Comment (elliottt) [reply main/realmain.cc]: I made those cases explitic, and raised the exception that was in the `default` case before.

**Pull request merged**